### DISCUSSION TITLE
Reconsider multiplayer ready state behavior

### DISCUSSION THREAD

**Discussion by yamplum** [ppy/osu#21966] [Ideas] [answered]
In multiplayer, if the host starts the game and a player did not click the "Ready" button, that player will be dropped into the spectator pool and will not get to play.

This seems unnecessarily punishing on new and old players alike as it's quite easy to forget to ready up. Personally, I often tab out of the game while waiting on the lobby and tab back in when I hear the song starting; if I didn't ready up, I will likely not notice until it's too late.

For hosts, an additional trip-up point is being able to start a game without readying up yourself via countdown.

The only benefit of the ready state seems to be indicating to the host that the player is not AFK. Since players do not interact with each other, there is no harm to the rest of the lobby in an AFK player joining the game.

Perhaps it would be more intuitive to let all players in the lobby join the game by default and only move them to spectators if explicitly requested?

**Comment by bdach** [comment]
To me this discussion reads more as "remove the ready system". Because you might as well at that point if it's only going to be a display thing that doesn't do anything.

Starting gameplay via countdown without being readied up as host was an explicit feature.

Countdowns also come with audible cues whenever the countdown is reaching zero. This should be what you listen for when tabbed out.

**Reply by yamplum** [reply]
> Starting gameplay via countdown without being readied up as host was an explicit feature.

Yes, I can see how it can be useful for e.g. tournament staff. It can be a confusing interaction at first though, because immediate start is locked behind ready state, so you begin to subconsciously associate "start game = host plays", which is incorrect in the case of a countdown start.

> Countdowns also come with audible cues whenever the countdown is reaching zero. This should be what you listen for when tabbed out.

As a host, this doesn't address the case described above. As a player, unfortunately, not all hosts extend the courtesy of counting down. Even in those cases, there is usually a fair bit of downtime at the start of a game (countdown, load the map, wait for other players to sync, possibly an intro) which prompts a delayed tab-in for me -- you can fit so many reddit posts in that bad boy ;)

One other thing that tripped me up is that the last three seconds of a countdown use really aggressive sound effects, which has caused me to panic thinking I wasn't ready and unready myself.

> To me this discussion reads more as "remove the ready system".

I chose my words carefully because I don't know the best path forward and I don't have strong preference either way, but I have tried out lazer multi today and have tripped up (and saw other people trip up) on these things. It's not good new user experience. You could say that one learns these things with experience, but I see it as an opportunity for improvement.

> Because you might as well at that point if it's only going to be a display thing that doesn't do anything.

One has to ask, what are the clear benefits of the ready system? Right now the only thing I see it do is punish users' inattention, which is worse than doing nothing. Does it really need to be in the game, or is it just the inertia of tradition? Is there a better way of doing things?

**Comment by peppy** [comment, marked as answer]
I think forcing all users which aren't in the spectate state to join the match is more correct. Not only because that's how stable does it, but it's more user-friendly. Let's change things to work in that direction.

Tracking via https://github.com/ppy/osu/issues/22090.
